modules/cert-manager-override-arguments.adoc

```diff
@@ -0,0 +1,94 @@
+// Module included in the following assemblies:
+//
+// * security/cert_manager_operator/cert-manager-customizing-api-fields.adoc
+
+:_content-type: PROCEDURE
+[id="cert-manager-override-arguments_{context}"]
+= Customizing cert-manager by overriding arguments from the cert-manager Operator API
+
+You can override the default arguments for the {cert-manager-operator} by adding a `spec.controllerConfig` section in the `CertManager` resource.
+
+.Prerequisites
+
+* You have access to the {product-title} cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Edit the `CertManager` resource by running the following command:
++
+[source,terminal]
+----
+$ oc edit certmanager cluster
+----
+
+. Add a `spec.controllerConfig` section with the following override arguments:
++
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1alpha1
+kind: CertManager
+metadata:
+  name: cluster
+  ...
+spec:
+  ...
+  controllerConfig: <1>
+    overrideArgs:
+      - '--dns01-recursive-nameservers=<host>:<port>' <2>
+      - '--dns01-recursive-nameservers-only' <3>
+      - '--acme-http01-solver-nameservers=<host>:<port>' <4>
+      - '--v=<verbosity_level>' <5>
+      - '--metrics-listen-address=<host>:<port>' <7>
+----
+<1> Add the `spec.controllerConfig` section.
+<2> Provide a comma-separated list of `<host>:<port>` nameservers to query for the DNS-01 self check. For example, `--dns01-recursive-nameservers=1.1.1.1:53`.
+<3> Specify to only use recursive nameservers instead of checking the authoritative nameservers associated with that domain.
+<4> Provide a comma-separated list of `<host>:<port>` nameservers to query for the ACME HTTP01 self check. For example, `--acme-http01-solver-nameservers=1.1.1.1:53`.
+<5> Specify to set the log level verbosity to determine the amount of detail in logged error messages.
+<6> Specify the host and port for the metrics endpoint. The default value is `--metrics-listen-address=0.0.0.0:9402`.
+
+. Save the file to apply the changes.
+
```

Comment:
Maybe you need add a note that, the supported overridden arguments are intentionally limited. Otherwise oc `get pod <redeployed_cert-manager_pod> -n cert-manager -o yaml` does show other arguments which can't be overridden, which can confuse users then.
See https://github.com/openshift/cert-manager-operator/blob/master/pkg/controller/deployment/deployment_overrides_validation.go#L19
@swghosh can have more suggestion if any.

Reply:
Sure, we can add a note about the currently supported arg list and that it is currently limited to include only the args that have support with our operator.
Users who wish to use beyond them can use `spec.unsupportedConfigOverrides` (which should be specified with warning/red, etc. as that is well **unsupported**).

Reply:
@swghosh I added the note as a warning.